### ISSUE TITLE
feat: Expand 'helm search'

### ISSUE THREAD

**Issue by technosophos** [kubernetes/helm#1199]
It has been suggested that we expand the scope of `helm search`, and add new similar tools, in an effort to improve findability of charts.

### New output for Helm search

No longer show versions as a default in search output:

```
$ helm search nginx
stable/nginx
stable/nginx-reverse-proxy
```

### Add a command: `helm charts list`

```
$ helm charts list
stable/alpine
stable/nginx
incubator/zookeeper
```

If `helm search` is called with no arguments, it returns the results of `helm charts list`

### List all charts in a specific repository with `helm charts list --repo stable`

```
$ helm charts list --repo stable
stable/alpine
stable/nginx
```

Alias `helm search --repo stable` to the above.

### List versions only when requested

```
$ helm search --versions nginx
stable/nginx      1.1.2
stable/nginx      1.2.3
```

**Comment by technosophos** [comment]
As of #1229:

- The command `helm search` (no args) returns a list of all charts
- `helm search stable/` (note the slash) returns all charts in the repo `stable`
- BUT charts are still listed by version number

**Comment by technosophos** [comment]
It has been requested that the `helm charts *` subcommands be moved to v2.1.0 so that we don't have to rush them right before Alpha.5. I think that all of the features listed above are now present (as of #1284). So when that gets merged, I will move the milestone on this to 2.1.0-Alpha.1 unless somebody objects.
